### PR TITLE
Security rework to synchronous

### PR DESCRIPTION
**Pull request by Verdent**
Here can be tracked progress of #5413

### CODE REVIEW

File: security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java [hunk 1]
```diff
@@ -50,40 +47,34 @@ static String fixServerType(String serverType) {
...
-                .connectTimeout(clientTimeout.toMillis(), TimeUnit.MILLISECONDS)
-                .readTimeout(clientTimeout.toMillis(), TimeUnit.MILLISECONDS)
-                .relativeUris(relativeUris);
+    static Http1Client.Http1ClientBuilder webClientBaseBuilder(String proxyProtocol,
+                                                               String proxyHost,
+                                                               int proxyPort,
+                                                               boolean relativeUris,
+                                                               Duration clientTimeout) {
+        Http1Client.Http1ClientBuilder webClientBuilder = WebClient.builder()
+                //TODO Níma client tracing
```
Comment (tomas-langer) [inline security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java:56]: Tracing is ready

File: security/providers/oidc/pom.xml [hunk 2]
```diff
@@ -101,16 +97,16 @@
...
-            <groupId>io.helidon.jersey</groupId>
-            <artifactId>helidon-jersey-client</artifactId>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.helidon.jersey</groupId>
-            <artifactId>helidon-jersey-media-jsonp</artifactId>
-            <scope>provided</scope>
-        </dependency>
+<!--        <dependency>-->
```
Comment (tomas-langer) [inline security/providers/oidc/pom.xml:100]: remove commented out text

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java [hunk 3]
```diff
@@ -50,40 +47,34 @@ static String fixServerType(String serverType) {
...
+                .mediaContext(MediaContext.builder()
+                                      .discoverServices(false)
+                                      .addMediaSupport(JsonpSupport.create(Config.empty()))
+                                      .build())
+                .channelOptions(SocketOptions.builder()
+                                        .connectTimeout(clientTimeout)
+                                        .readTimeout(clientTimeout)
+                                        .build());
+//        TODO Níma relative uris
+//                .relativeUris(relativeUris);
```
[the remark this answers is not in the document]
Comment (Verdent) [reply security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java:67]: Fixed

File: security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java (hunk 1 above)
Comment (Verdent) [reply security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcUtil.java:56]: Fixed

File: security/providers/oidc/pom.xml (hunk 2 above)
Comment (Verdent) [reply security/providers/oidc/pom.xml:100]: fixed

File: security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/Tenant.java [hunk 4]
```diff
@@ -133,12 +120,15 @@ public static Tenant create(OidcConfig oidcConfig, TenantConfig tenantConfig) {
                     .addOutboundSecurityProvider(httpBasicAuth)
                     .build();
 
-            webClientBuilder.addService(WebClientSecurity.create(tokenOutboundSecurity));
+            //TODO Níma client security?
```
[the remark this answers is not in the document]
Comment (Verdent) [reply security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/Tenant.java:123]: Follow up issue https://github.com/helidon-io/helidon/issues/6967

**Comment by Verdent** [comment]
Created two follow up issues: https://github.com/helidon-io/helidon/issues/6967 and https://github.com/helidon-io/helidon/issues/6968

File: security/providers/idcs-mapper/src/main/java/io/helidon/security/providers/idcs/mapper/IdcsRoleMapperProvider.java [hunk 5]
```diff
@@ -36,29 +44,23 @@
...
     private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
 
     private final EvictableCache<String, List<Grant>> roleCache;
-    private final WebTarget assertEndpoint;
+    private final String asserterUri;
+    private final URI tokenEndpointUri;
 
     // caching application token (as that can be re-used for group requests)
-    private final AppToken appToken;
+    private final AppTokenRx appToken;
```
Comment (tomas-langer) [inline security/providers/idcs-mapper/src/main/java/io/helidon/security/providers/idcs/mapper/IdcsRoleMapperProvider.java:60]: `AppTokenRx` name should not be used.

Comment (Verdent) [reply security/providers/idcs-mapper/src/main/java/io/helidon/security/providers/idcs/mapper/IdcsRoleMapperProvider.java]: fixed

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**